ts/components/conversation/right-panel/RightPanel.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { useSelector } from 'react-redux';
+import { getRightOverlayMode } from '../../../state/selectors/section';
+import { checkIsFeatureReleased } from '../../../util/releaseFeature';
+import { OverlayDisappearingMessages } from './overlay/disappearing-messages/OverlayDisappearingMessages';
+import { OverlayRightPanelSettings } from './overlay/OverlayRightPanelSettings';
+
+const ClosableOverlay = () => {
+  const rightOverlayMode = useSelector(getRightOverlayMode);
+  const [showNewDisppearingMessageModes, setShowNewDisppearingMessageModes] = useState(false);
+
+  const checkForFeatureRelease = useCallback(async () => {
+    const isReleased = await checkIsFeatureReleased('Disappearing Messages V2');
+    return isReleased;
+  }, []);
+
+  useEffect(() => {
+    let isCancelled = false;
+    checkForFeatureRelease()
+      .then(result => {
+        if (isCancelled) {
+          return;
+        }
+        setShowNewDisppearingMessageModes(result);
+      })
+      .catch(() => {
+        if (isCancelled) {
+          return;
+        }
+      });
+
+    return () => {
+      isCancelled = true;
+    };
+  }, [checkForFeatureRelease]);
```

Comment:
not sure what this is supposed to do, but that `useCallback` has no dependency so is always the same.
then the useEffect has just that `checkForFeatureRelease` dependency which is always the same too as got from the useCallback above.
This looks to me to just


```ts
  useEffect(() => {
    let isCancelled = false;

    checkIsFeatureReleased('Disappearing Messages V2')
      .then(result => {
        if (isCancelled) {
          return;
        }
        setShowNewDisppearingMessageModes(result);
      })
      .catch(() => {
        if (isCancelled) {
          return;
        }
      });

    return () => {
      isCancelled = true;
    };
  }, []);
```
no?

Reply:
I think you are correct. I probably wrote the `useCallback` out of habit to prevent linting errors. Will update